Peer/Program.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Peer.ConnectorApi;
+using Peer.Connectors;
+using Peer.Domain;
+using Peer.Domain.Models;
+using Peer.IO;
+using Peer.User;
+
+namespace Peer
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            IServiceCollection services = new ServiceCollection();
+            var configModel = BuildConfig();
+
+            RegisterServices(services, configModel);
+            PeerPullRequestAPI<GithubPullRequestSource> GitpRListApi;
+            PeerPullRequestAPI<AzdoPullRequestSource> AzDopRListApi;
+            PeerPullRequestAPI<GitlabPullRequestSource> GitlabpRListApi;
+            BuildServices(services, out GitpRListApi, out AzDopRListApi, out GitlabpRListApi);
+
+            _ = await GitpRListApi.GetPullRequests();
+            _ = await AzDopRListApi.GetPullRequests();
+            _ = await GitlabpRListApi.GetPullRequests();
+
+            Console.ReadLine();
+        }
+
+        private static AppConfig BuildConfig()
+        {
+            var config = new ConfigurationBuilder()
+                            .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                            .AddEnvironmentVariables()
+                            .Build();
+            var configModel = config.GetSection("UserConfig").Get<AppConfig>();
+            return configModel;
+        }
+
+        private static void BuildServices(IServiceCollection services, out PeerPullRequestAPI<GithubPullRequestSource> GitpRListApi, out PeerPullRequestAPI<AzdoPullRequestSource> AzDopRListApi, out PeerPullRequestAPI<GitlabPullRequestSource> GitlabpRListApi)
+        {
+            GitpRListApi = services
```

Comment:
You should only ever build the service provider once. In this case if you've registered singletons that are used in both AzDO/GitLab you'll have 2 instances of them instead of the 1 you expect since singletons are one per *service provider*

Reply:
Done